pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -569,6 +571,20 @@ def log(message: str, *, warning: bool = False) -> None:
                 f'Extending current KCFG with the following: {message}{result_info_message}',
             )
 
+        def optimize_step(
+            cterm: CTerm, depth: int, next_node_logs: tuple[LogEntry, ...], rule_labels: list[str]
+        ) -> bool:
+            in_edges = self.edges(target_id=node.id)
+            if len(in_edges) == 1:
+                in_edge = in_edges[0]
+                self.remove_edge(in_edge.source.id, node.id)
+                self.let_node(node_id=node.id, cterm=cterm)
+                self.create_edge(in_edge.source.id, node.id, in_edge.depth + depth, list(in_edge.rules) + rule_labels)
+                logs[node.id] = logs[node.id] + next_node_logs if node.id in logs else next_node_logs
+                log(f'basic block at depth {depth}: update: {node.id}')
+                return True
+            return False
```

Comment:
Only remaining comment is that I would just inline this into where `Step` is handled, rather than have a separate function, since it's only going to be used in that one spot.

Reply:
Ok, the only reason I separated it is because I thought we might want to do other optimisations for other non-`Step` variants and I wanted a modular way of capturing this. Happy to inline.

Reply:
Inlined now.